editors/emacs/swarm-mode.el

```diff
@@ -7,151 +7,170 @@
-               "undefined"
-               "fail"
-               "not"
-               "format"
-               "chars"
-               "split"
-               "charat"
-               "tochar"
-               "key"
-             ))
-             (x-commands '(
-               "noop"
-               "wait"
-               "selfdestruct"
-               "move"
-               "backup"
-               "volume"
-               "path"
-               "push"
-               "stride"
-               "turn"
-               "grab"
-               "harvest"
-               "sow"
-               "ignite"
-               "place"
-               "ping"
-               "give"
-               "equip"
-               "unequip"
-               "make"
-               "has"
-               "equipped"
-               "count"
-               "drill"
-               "use"
-               "build"
-               "salvage"
-               "reprogram"
-               "say"
-               "listen"
-               "log"
-               "view"
-               "appear"
-               "create"
-               "halt"
-               "time"
-               "scout"
-               "whereami"
-               "waypoint"
-               "structure"
-               "floorplan"
-               "hastag"
-               "tagmembers"
-               "detect"
-               "resonate"
-               "density"
-               "sniff"
-               "chirp"
-               "watch"
-               "surveil"
-               "heading"
-               "blocked"
-               "scan"
-               "upload"
-               "ishere"
-               "isempty"
-               "meet"
-               "meetall"
-               "whoami"
-               "setname"
-               "random"
-               "run"
-               "return"
-               "try"
-               "swap"
-               "atomic"
-               "instant"
-               "installkeyhandler"
-               "teleport"
-               "as"
-               "robotnamed"
-               "robotnumbered"
-               "knows"
-               "east"
-               "north"
-               "west"
-               "south"
-               "down"
-               "forward"
-               "left"
-               "back"
-               "right"
-             ))
-             (x-types '("int" "text" "dir" "bool" "cmd" "void" "unit" "actor"))
+(defvar swarm-mode-types-regexp
+   "\\b[A-Z][a-zA-Z_]*\\b"
+  "Regexp that recognizes types (all uppercase strings are supposed to be types) in swarm language.")
 
-             (x-keywords-regexp (regexp-opt x-keywords 'words))
-             (x-builtins-regexp (regexp-opt x-builtins 'words))
-             (x-commands-regexp (regexp-opt x-commands 'words))
-             (x-types-regexp (regexp-opt x-types 'words)))
+(defvar swarm-mode-keywords-regexp
+  (concat "\\b" (regexp-opt '("def" "tydef" "end" "let" "in" "require") t) "\\b")
+  "Regexp that recognizes types in swarm language.")
```

Comment:
Should this say "keywords" instead of "types"?

Reply:
Right!